src/outputs/alsa.c

```diff
@@ -133,15 +136,18 @@ prebuf_free(struct alsa_session *as)
 static void
 alsa_session_free(struct alsa_session *as)
 {
-  if (!as)
-    return;
+  if (as)
```

Comment:
I don't understand this change. It prefer returning quickly if !as. Also, it is fine to free null pointers (it's a no-op), so you don't need the if's.